### PR TITLE
HOSS2-98-Update unit test coverage and negative tests

### PR DESCRIPTION
**Pull request by AdamCoops**

### CODE REVIEW

File: test/uk/gov/hmrc/homeofficesettledstatus/controllers/StatusCheckByNinoRequestFormSpec.scala [hunk 1]
```diff
@@ -48,94 +49,338 @@ class StatusCheckByNinoRequestFormSpec extends AnyWordSpecLike with Matchers wit
...
+        val input = formInput.updated("nino", "invalid")
+        form.bind(input).errors shouldBe List(FormError("nino", "error.nino.invalid-format"))
+      }
 
-    "report an error when givenName is invalid" in {
-      val form = HomeOfficeSettledStatusFrontendController.StatusCheckByNinoRequestForm
-      val input = formInput.updated("givenName", "11267162")
-      form.bind(input).errors shouldBe List(FormError("givenName", "error.givenName.invalid-format"))
-    }
+      "report an error when NINO is an string invalid by generated range" in {
```
Comment (liampan) [inline test/uk/gov/hmrc/homeofficesettledstatus/controllers/StatusCheckByNinoRequestFormSpec.scala:66]: Are we testing code we dont own here? I think we are using the hmrc Nino lib to validate these

File: test/uk/gov/hmrc/homeofficesettledstatus/controllers/StatusCheckByNinoRequestFormSpec.scala [hunk 2]
```diff
@@ -48,94 +49,338 @@ class StatusCheckByNinoRequestFormSpec extends AnyWordSpecLike with Matchers wit
...
+      }
     }
 
-    "disallow empty dateOfBirth.day" in {
-      val form = HomeOfficeSettledStatusFrontendController.StatusCheckByNinoRequestForm
-      val input = formInput.updated("dateOfBirth.day", "")
-      form.bind(input).value shouldBe None
+    "First Name tests" should {
+
+      "report an error when givenName is missing" in {
```
Comment (liampan) [inline test/uk/gov/hmrc/homeofficesettledstatus/controllers/StatusCheckByNinoRequestFormSpec.scala:193]: These look good. I think we could create a Gen that would incorporate a lot of them together if you wanted. Up to you though :)

File: test/uk/gov/hmrc/homeofficesettledstatus/controllers/StatusCheckByNinoRequestFormSpec.scala [hunk 3]
```diff
@@ -48,94 +49,338 @@ class StatusCheckByNinoRequestFormSpec extends AnyWordSpecLike with Matchers wit
...
+      }
     }
 
-    "report an error when empty dateOfBirth.month " in {
-      val form = HomeOfficeSettledStatusFrontendController.StatusCheckByNinoRequestForm
-      val input = formInput.updated("dateOfBirth.month", "")
-      form.bind(input).errors shouldBe List(FormError("dateOfBirth", "error.dateOfBirth.invalid-format"))
+    "Family Name Tests" should {
+
+      "report an error when familyName is missing" in {
```
Comment (liampan) [inline test/uk/gov/hmrc/homeofficesettledstatus/controllers/StatusCheckByNinoRequestFormSpec.scala:233]: As these tests are duplicate of the given name is it worth making them a method where "givenName" or "familyName" could be passed in ?

File: project/AppDependencies.scala [hunk 4]
```diff
@@ -21,6 +21,8 @@ object AppDependencies {
   private val test: Seq[ModuleID] = Seq(
     "org.scalatest"               %% "scalatest"            % "3.2.9",
     "org.mockito"                 %  "mockito-core"         % "3.5.15",
+    "org.scalatestplus"          %% "scalatestplus-scalacheck" % "3.1.0.0-RC2",
```
Comment (liampan) [inline project/AppDependencies.scala:24]: alignment here.

**Review by liampan — COMMENTED** [review]
Looks good. 😊 
I think we could refactor in a few places, however happy if we wanna do that separately when the test coverage is in place :)

File: test/uk/gov/hmrc/homeofficesettledstatus/controllers/StatusCheckByNinoRequestFormSpec.scala (hunk 2 above)
Comment (AdamCoops) [reply test/uk/gov/hmrc/homeofficesettledstatus/controllers/StatusCheckByNinoRequestFormSpec.scala:193]: Yeah, so I originally did that but I also think it's probably better to separate out each case so we know exactly where the fails are, right?

Comment (AdamCoops) [reply test/uk/gov/hmrc/homeofficesettledstatus/controllers/StatusCheckByNinoRequestFormSpec.scala:193]: Could do both easily enough

File: test/uk/gov/hmrc/homeofficesettledstatus/controllers/StatusCheckByNinoRequestFormSpec.scala (hunk 1 above)
Comment (AdamCoops) [reply test/uk/gov/hmrc/homeofficesettledstatus/controllers/StatusCheckByNinoRequestFormSpec.scala:66]: Nah, it's just a list of all the ways that the form can fall over.

File: test/uk/gov/hmrc/homeofficesettledstatus/controllers/StatusCheckByNinoRequestFormSpec.scala (hunk 2 above)
Comment (AdamCoops) [reply test/uk/gov/hmrc/homeofficesettledstatus/controllers/StatusCheckByNinoRequestFormSpec.scala:193]: Nah, I think it's redundant personally. happy to have a chat about it tho.

File: test/uk/gov/hmrc/homeofficesettledstatus/controllers/StatusCheckByNinoRequestFormSpec.scala (hunk 3 above)
Comment (AdamCoops) [reply test/uk/gov/hmrc/homeofficesettledstatus/controllers/StatusCheckByNinoRequestFormSpec.scala:233]: Some of them are duplicates, yeah. There are some differences those as the min length and accepted characters etc follow different validation.

**Review by burns238 — APPROVED** [review]

**Pull request merged**